test/integration/node/lifecycle_test.go

```diff
@@ -290,175 +284,94 @@ func TestTaintBasedEvictions(t *testing.T) {
 				true,             // Run taint manager
 			)
 			if err != nil {
-				t.Errorf("Failed to create node controller: %v", err)
-				return
+				t.Fatalf("Failed to create node controller: %v", err)
 			}
 
 			// Waiting for all controllers to sync
 			externalInformers.Start(testCtx.Ctx.Done())
 			externalInformers.WaitForCacheSync(testCtx.Ctx.Done())
-			testutils.SyncInformerFactory(testCtx)
 
-			// Run all controllers
+			// Run the controller
 			go nc.Run(testCtx.Ctx)
-			go testCtx.Scheduler.Run(testCtx.Ctx)
 
 			nodeRes := v1.ResourceList{
 				v1.ResourceCPU:    resource.MustParse("4000m"),
 				v1.ResourceMemory: resource.MustParse("16Gi"),
 				v1.ResourcePods:   resource.MustParse("110"),
 			}
 
+			nodeIndex := 1 // the exact node doesn't matter, pick one
 			var nodes []*v1.Node
 			for i := 0; i < nodeCount; i++ {
-				nodes = append(nodes, &v1.Node{
+				node := &v1.Node{
 					ObjectMeta: metav1.ObjectMeta{
-						Name:   fmt.Sprintf("node-%d", i),
-						Labels: map[string]string{v1.LabelTopologyRegion: "region1", v1.LabelTopologyZone: "zone1"},
+						Name: fmt.Sprintf("node-%d", i),
+						Labels: map[string]string{
+							v1.LabelTopologyRegion:                  "region1",
+							v1.LabelTopologyZone:                    "zone1",
+							"node.kubernetes.io/exclude-disruption": "true",
+						},
 					},
 					Spec: v1.NodeSpec{},
 					Status: v1.NodeStatus{
 						Capacity:    nodeRes,
 						Allocatable: nodeRes,
-						Conditions: []v1.NodeCondition{
-							{
-								Type:              v1.NodeReady,
-								Status:            v1.ConditionTrue,
-								LastHeartbeatTime: metav1.Now(),
-							},
-						},
 					},
-				})
-				if _, err := cs.CoreV1().Nodes().Create(context.TODO(), nodes[i], metav1.CreateOptions{}); err != nil {
-					t.Errorf("Failed to create node, err: %v", err)
+				}
+				if i == nodeIndex {
+					node.Status.Conditions = append(node.Status.Conditions, test.nodeConditions...)
+				} else {
+					node.Status.Conditions = append(node.Status.Conditions, v1.NodeCondition{
+						Type:   v1.NodeReady,
+						Status: v1.ConditionTrue,
+					})
+				}
+				nodes = append(nodes, node)
+				if _, err := cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{}); err != nil {
+					t.Fatalf("Failed to create node: %q, err: %v", klog.KObj(node), err)
 				}
 			}
 
-			neededNode := nodes[1]
 			if test.pod != nil {
+				test.pod.Spec.NodeName = nodes[nodeIndex].Name
 				test.pod.Name = fmt.Sprintf("testpod-%d", i)
 				if len(test.pod.Spec.Tolerations) > 0 {
-					test.pod.Spec.Tolerations[0].TolerationSeconds = &tolerationSeconds[i]
```

Comment:
do we need the `i` in line 253 now?

> 	for i, test := range tests {

there are too many `i` in this test :shrug:

Reply:
Agree, renamed the outer `i` as `testIndex`